### PR TITLE
feat(apollo): sync users, delete a user, pagination and add url param

### PR DESCRIPTION
**Pull request by mbliakharskyi**
## Description

Apollo Integration
1. Update the codebase
2. Implemented pagination
3. Modified user deletion
4. Added the `url` param to ElbaUser

## Additional Notes

(*) Not added the `isSuspendable `param to ElbaUser

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/apollo/src/app/install/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import type { Metadata } from 'next';
+import '@elba-security/design-system/dist/styles.css';
+
+export { SetupLayout as default } from '@elba-security/design-system';
+
+export const metadata: Metadata = {
+  title: 'Connect elba to Fivertran',
+  description: "Official elba's Fivertran integration",
```
Comment (llanicm) [inline apps/apollo/src/app/install/layout.tsx:8]:
```suggestion
  title: 'Connect elba to Apollo',
  description: "Official elba's Apollo integration",
```

File: apps/apollo/src/connectors/apollo/users.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  userId: string;
+  apiKey: string;
+};
+
+const perPage = env.APOLLO_USERS_SYNC_BATCH_SIZE;
+
+export const getUsers = async ({ apiKey, after }: GetUsersParams) => {
+  const url = new URL(`${env.APOLLO_API_BASE_URL}/v1/users/search`);
+
+  url.searchParams.append('per_page', String(perPage));
```
Comment (llanicm) [inline apps/apollo/src/connectors/apollo/users.ts:37]:
```suggestion
  url.searchParams.append('per_page', String(env.APOLLO_USERS_SYNC_BATCH_SIZE));
```

File: apps/apollo/src/connectors/apollo/users.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  apiKey: string;
+  after: number;
+};
+
+export type DeleteUserParams = {
+  userId: string;
+  apiKey: string;
+};
+
+const perPage = env.APOLLO_USERS_SYNC_BATCH_SIZE;
```
Comment (llanicm) [inline apps/apollo/src/connectors/apollo/users.ts:32]:
```suggestion

```

File: apps/apollo/src/connectors/apollo/users.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+export const getUsers = async ({ apiKey, after }: GetUsersParams) => {
```
Comment (llanicm) [inline apps/apollo/src/connectors/apollo/users.ts:34]:
```suggestion
export const getUsers = async ({ apiKey, page }: GetUsersParams) => {
```

File: apps/apollo/src/inngest/functions/users/delete-user.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import { deleteUser as deleteApolloUser } from '@/connectors/apollo/users';
+import { env } from '@/common/env';
+import { decrypt } from '@/common/crypto';
+
+export const deleteUser = inngest.createFunction(
+  {
+    id: 'apollo-delete-user',
+    concurrency: {
+      key: 'event.data.organisationId',
+      limit: env.APOLLO_DELETE_USER_CONCURRENCY,
```
Comment (llanicm) [inline apps/apollo/src/inngest/functions/users/delete-user.ts:15]: How come this has it own concurrency?

File: apps/apollo/src/inngest/functions/users/sync-users.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+import { type ApolloUser } from '@/connectors/apollo/users';
+import { createElbaClient } from '@/connectors/elba/client';
+import { decrypt } from '@/common/crypto';
+
+const formatElbaUser = (user: ApolloUser): User => ({
+  id: user.id,
+  displayName: user.name || user.email,
+  email: user.email,
+  additionalEmails: [],
+  url: `https://app.apollo.io/#/users/${user.id}/edit`,
```
Comment (llanicm) [inline apps/apollo/src/inngest/functions/users/sync-users.ts:18]: isSuspendable is missing here

Comment (mbliakharskyi) [reply apps/apollo/src/inngest/functions/users/sync-users.ts:18]: Yes, as I mentioned in the pr, I couldn't filter the owner who created API key in the users list.
And also couldn't find any specific endpoints to get the owner id.

Comment (llanicm) [reply apps/apollo/src/inngest/functions/users/sync-users.ts:18]: Yeah sorry I added this comment to the review before @athielba passed on the message :)

**Pull request merged**